.setup/bin/setup_sample_courses.py

```diff
@@ -1217,6 +1218,85 @@ def add_sample_forum_data(self):
                               type=postData[9],
                               has_attachment=True if postData[10] != "f" else False)
 
+    def add_sample_polls_data(self):
+        # set sample course to have polls enabled by default
+        course_json_file = os.path.join(self.course_path, 'config', 'config.json')
+        with open(course_json_file, 'r+') as open_file:
+            course_json = json.load(open_file)
+            course_json['course_details']['polls_enabled'] = True
+            open_file.seek(0)
+            open_file.truncate()
+            json.dump(course_json, open_file, indent=2)
+
+        # load the sample polls from input file
+        polls_data_path = os.path.join(SETUP_DATA_PATH, "polls", "polls_data.json")
+        with open(polls_data_path, 'r') as polls_file:
+            polls_data = json.load(polls_file)
+
+        # set some values that depend on current time
+        polls_data[0]["image_path"] = self.course_path + polls_data[0]["image_path"]
+        polls_data[2]["release_date"] = f"{datetime.today().date()}"
+
+        # add attached image
+        image_dir = os.path.dirname(polls_data[0]["image_path"])
+        if os.path.isdir(image_dir):
+            shutil.rmtree(image_dir)
+
+        os.makedirs(image_dir)
+        os.system(f"chown -R submitty_php:sample_tas_www {image_dir}")
+        copyfile(os.path.join(SETUP_DATA_PATH, "polls", "sea_animals.png"), polls_data[0]["image_path"])
+
+        # add polls to DB
+        polls_table = Table("polls", self.metadata, autoload=True)
+        poll_options_table = Table("poll_options", self.metadata, autoload=True)
+        poll_responses_table = Table("poll_responses", self.metadata, autoload=True)
+
+        for poll in polls_data:
+            self.conn.execute(polls_table.insert(),
+                              name=poll["name"],
+                              question=poll["question"],
+                              status=poll["status"],
+                              release_date=poll["release_date"],
+                              image_path=poll["image_path"],
+                              question_type=poll["question_type"])
+            i = 0
+            for poll_response in poll["responses"]:
+                self.conn.execute(poll_options_table.insert(),
+                                  option_id=i,
+                                  order_id=i,
+                                  poll_id=poll["id"],
+                                  response=poll_response,
+                                  correct=(i in poll["correct_responses"]))
+                i += 1
```

Comment:
Given that `poll_response` is used just once and so not saving a bunch of typing on doing it instead of `poll["responses"][i]`, better (in my opinion) to just have this be a regular counting indexed loop as allows getting rid of the two surrounding lines.
```suggestion
            for i in range(len(poll["responses"])):
                self.conn.execute(poll_options_table.insert(),
                                  option_id=i,
                                  order_id=i,
                                  poll_id=poll["id"],
                                  response=poll["responses"][i],
                                  correct=(i in poll["correct_responses"]))
```